utils/docker-ce/files/dockerd.init

```diff
@@ -3,14 +3,101 @@
 USE_PROCD=1
 START=25
 
+EXTRA_COMMANDS="uciadd ucidel"
+EXTRA_HELP="\
+	uciadd  Add default bridge configuration to network and firewall uci config
+	ucidel  Delete default bridge configuration from network and firewall uci config"
+
 DOCKERD_CONF="/tmp/dockerd/daemon.json"
 
+uci_quiet() {
+	uci -q ${@} >/dev/null
+}
+
 json_add_array_string() {
 	json_add_string "" "$1"
 }
 
+boot() {
+	uciadd
+	rc_procd start_service
+}
+
+uciupdate() {
+	local net="$1"
+
+	uci -q get network.docker >/dev/null || {
+		logger -t "dockerd-init" -p warn "No network uci config section for docker default bridge (docker0) found"
+		return
+	}
+
+	[ -z "$net" ] && {
+		logger -t "dockerd-init" -p notice "Removing network uci config options for docker default bridge (docker0)"
+		uci_quiet delete network.docker.netmask
+		uci_quiet delete network.docker.ipaddr
+		uci_quiet commit network
+		return
+	}
+
+	eval "$(ipcalc.sh "$net")"
+	logger -t "dockerd-init" -p notice "Updating network uci config option \"$net\" for docker default bridge (docker0)"
+	uci_quiet set network.docker.netmask="$NETMASK"
+	uci_quiet set network.docker.ipaddr="$IP"
+	uci_quiet commit network
+}
+
+uciadd() {
+	/etc/init.d/dockerd running && {
+		echo "Please stop dockerd service first"
+		exit 0
+	}
+
+	# Add network interface
+	if ! uci -q get network.docker >/dev/null; then
+		logger -t "dockerd-init" -p notice "Adding docker default bridge to network uci config (docker0)"
+		uci_quiet add network interface
+		uci_quiet rename network.@interface[-1]="docker"
+		uci_quiet set network.docker.ifname="docker0"
+		uci_quiet set network.docker.proto="static"
+		uci_quiet set network.docker.auto="0"
+		uci_quiet commit network
+	fi
+
+	# Add firewall zone
+	if ! uci -q get firewall.docker >/dev/null; then
+		logger -t "dockerd-init" -p notice "Adding docker default bridge firewall zone (docker0)"
+		uci_quiet add firewall zone
+		uci_quiet rename firewall.@zone[-1]="docker"
+		uci_quiet set firewall.docker.network="docker"
+		uci_quiet set firewall.docker.input="REJECT"
```

Comment:
@feckert, I've been trying to have [docker use the local DNS server by default](https://github.com/openwrt/packages/commit/c21cfa44d44807803535c0e35a78a1540e4546e0) but I've noticed the traffic is being blocked on the input zone. 
It seems to me now that we have the safety of the `blocked_interfaces` preventing someone accidentally opening a port on the `wan`, as a default. 
We should be able to treat `docker0` as any other local interface and not need a firewall rule by default, or perhaps treat it like another `lan` zone.
I'd be interested to hear your thoughts or if you recall anything about the original intention of this.

Reply:
I have a PR draft for this https://github.com/openwrt/packages/pull/12897#discussion_r579615391

Reply:
@G-M0N3Y-2503 I wanted the docker0 interface to be in its own firewall zone. Until now, it was possible for all interfaces, including the wan interface, to access the docker containers without a firewall controller. Until now, there have been two methods to control the docker interface handling.
* With the option **blocked_interfaces** ( no integration into the fw3 )
* And then other via **/etc/init.d/docker uciadd \<interface\>**

I always use the second one via uciadd. This integrates my docker interfaces/firewwallzone and thus the containers into the OpenWrt firewall handling. With REJECT I also wanted to control access from LAN to the Conteiner and forwarding should also be prevented. If someone wants to access your container, they have to open the port in the firewall the container is listening on. I therefore did not want to put the docker interface in the lan zone. It may be that not everyone should access the container. I wanted to think of it more as a different network.

Reply:
In researching what an expected default might be I found that docker already has [Integration with Firewalld](https://docs.docker.com/network/iptables/#integration-with-firewalld). It also appears that the zone is [set to `ACCEPT` by default](https://github.com/moby/libnetwork/blob/fa125a3512ee0f6187721c88582bf8c4378bd4d7/iptables/firewalld.go#L204). This default appears to be more like [my proposed changed](https://github.com/openwrt/packages/pull/14807), if I'm interpreting [the setting](https://firewalld.org/documentation/zone/options.html) correctly.

I tend to think that most people will primarily be running containers for services either for their LAN or WAN running them in OpenWrt. But I also agree that we should be mindful of the WAN given that it's atypical of other places docker might run. Given that there is this precedent set upstream, I'm inclined to think that we should be more allowing for people to quickly set up a service for their LAN.

With the [proposed changes](https://github.com/openwrt/packages/pull/14807) I've tested and containers have no WAN access in or out with fw3 or docker with iptables just on device and LAN access.

Reply:
@G-M0N3Y-2503 Ok from my side. I can also live with that :+1: